src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -345,52 +345,55 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs e)
         protected override void OnKeyDown(KeyEventArgs e)
         {
             var key = e.Key;
-
-            if (key == Key.Space || key == Key.Enter)
+            if (IsFocused)
             {
-                _isKeyboardPressed = true;
-                UpdatePseudoClasses();
+                if (key == Key.Space || key == Key.Enter)
+                {
+                    _isKeyboardPressed = true;
+                    UpdatePseudoClasses();
+                }
             }
-
             base.OnKeyDown(e);
         }
 
         /// <inheritdoc/>
         protected override void OnKeyUp(KeyEventArgs e)
         {
-            var key = e.Key;
```

Comment:
For both of these changes it's a lot easier to read the changes if you just check and exit on entry:

```cs
if (!IsFocused)
{
    base.OnKeyUp(e);
    return;
}
```

 1. That would remove a lot of noise in this PR 
 2. It helps with maintainability and readability to reduce nesting in the logic

Reply:
I preferred this solution because IL is more compact, see [example](https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgYGYAE9MCFMDeqmJ2WuAFAJTGlEqmOa0kC+LmADgE4D2wAUwDGggCaYAbmG7AArgEMI2ACyYA8gDsA0gICeAVU4UAohIEbgAQW4BzAM6Z5tu1UId2DUhwyYARr14lAEk7ADFeIVk7AXECTBsBYABuLm4wCXlBTGjkzFZMAF5MYG5ZASTUD1QfHABhTBA8N08SHn5hMUxeM2400QEVdW09QxMzC2t7R2caFuamUjAAMwoAQhDwyOjRWYWSej2mX3logDpNHQMjJ3sqCrnDmAB2e8OPQ+WKG7tTgBVeADKJTAGhs1AKBQARAANSG7PYHQ6kGAIACcFEhAAkBBAILw4a89u89sczhcRtcZoS2JVUNUsDgACINJqI1p8QQiGJdHp9AYwVTkq5jcxWZzTW4cNlMT4bCJRGLwhbSj4rb5/QHA0HgqGwpVIlVIkgo9FYnF4gkcN5W0jEhakgTnYbC753dyVIA===).

Reply:
I don't think IL really matters here, as JIT might produce very similarly efficient code in the end.
But I don't like doubling `base.OnKeyUp(e);` call. Early returns are great, but duplicating base calls is not so much.